### PR TITLE
Fixed Zipcode validation bug

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
On a school profile page, a user is able to select if they will be attending class at the school's main campus or at another location. When the user indicates that they will be taking classes at an "other" location, a field appears in which the user will enter a postal code. Currently, the field will appear accompanied by an error message before the user makes an entry. The field also accepts letters and special characters, but should only accept numbers.

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19650

## Testing done
Local Testing

## Screenshots
ezgif-3-afcb73f3cfbc

## Acceptance criteria
- [x] Zip code validation error only displays when bad input is entered

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by zurbergram — APPROVED** [review]
LGTM

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Check for empty string first.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/gi/reducers/calculator.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   FETCH_PROFILE_SUCCEEDED,
 } from '../actions';
 
-const beneficiaryZIPRegExTester = /\b\d{1,5}\b/;
+const beneficiaryZIPRegExTester = /^\d{1,5}$|^\d{1,5}-\d{4}$/;
```
Comment (Mottie) [inline src/applications/gi/reducers/calculator.js]: Pardon the new guy budding in. This regex will match zipcodes less than 5 digits long. If it's not validating while the user is typing, then it should only match 5 digits.

**Review by zurbergram — APPROVED** [review]
LGTM

File: src/applications/gi/reducers/calculator.js [hunk 2]
```diff
@@ -228,7 +227,7 @@ export default function(state = INITIAL_STATE, action) {
 
       let beneficiaryZIPError;
 
-      if (!beneficiaryZIPRegExTester.exec(beneficiaryZIP)) {
+      if (beneficiaryZIP.length > 5) {
```
Comment (Mottie) [inline src/applications/gi/reducers/calculator.js]: Shouldn't this be `.length < 5` since the number input will prevent numbers > 5 in length?

Comment (ksantiagomorgan6) [reply src/applications/gi/reducers/calculator.js]: Hello Mottie, this PR is still in draft I'm just testing some different things out. As for this if statement I want to throw the error if the length is greater than 5.

**Review by sonntag-adam — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**